### PR TITLE
Upgrade Kafka dependency to 2.4

### PR DESCRIPTION
**Pull request by viktorsomogyi**
Addresses the issue #1104 

* Added I0Itec zkclient dependency as it has been removed from Kafka
* `KafkaZkClient.getAllTopicsInCluster`'s return type changed from Seq to Set
* `AdminZkClient.addPartition`'s `existingAssignment`'s parameter type changed

### CODE REVIEW

File: cruise-control/src/main/scala/com/linkedin/kafka/cruisecontrol/executor/ExecutorUtils.scala [hunk 1]
```diff
@@ -131,8 +131,4 @@ object ExecutorUtils {
   def getAllLiveBrokerIdsInCluster(kafkaZkClient: KafkaZkClient): java.util.List[java.lang.Integer] = {
     seqAsJavaList(kafkaZkClient.getAllBrokersInCluster.map(_.id : java.lang.Integer))
   }
-
-  def getAllTopicsInCluster(kafkaZkClient: KafkaZkClient): java.util.List[String] = {
```
Comment (viktorsomogyi) [inline cruise-control/src/main/scala/com/linkedin/kafka/cruisecontrol/executor/ExecutorUtils.scala]: This method isn't called anywhere in the project so I removed it.

**Comment by viktorsomogyi** [comment]
@efeg would you please review this?

File: cruise-control/src/main/scala/com/linkedin/kafka/cruisecontrol/executor/ExecutorUtils.scala (hunk 1 above)
Comment (efeg) [reply cruise-control/src/main/scala/com/linkedin/kafka/cruisecontrol/executor/ExecutorUtils.scala]: Thanks for removing the dead code.

**Comment by viktorsomogyi** [comment]
@efeg ok, made the change with `close`. 

You mentioned that it's a requirement to compile CC with different Kafka versions, but between 2.3 and 2.4 `AdminZkClient.addPartitions` changed its parameter types so after making that change which is made here in this PR, it makes CC uncompilable with 2.3. Do you think we should find a workaround for this? I'm afraid there might not be much more option than duplicating some of the code of `AdminZkClient` and implement `addPartitions`.

**Comment by efeg** [comment]
> @efeg ok, made the change with `close`.
> 
> You mentioned that it's a requirement to compile CC with different Kafka versions, but between 2.3 and 2.4 `AdminZkClient.addPartitions` changed its parameter types so after making that change which is made here in this PR, it makes CC uncompilable with 2.3. Do you think we should find a workaround for this? I'm afraid there might not be much more option than duplicating some of the code of `AdminZkClient` and implement `addPartitions`.

@viktorsomogyi Backwards incompatible changes in this PR seem to be limited to `KafkaSampleStore`, which is a pluggable component. Presumably, such implementation changes in pluggable components should be acceptable as those, who compile CC with different Kafka versions (e.g. `pre-Kafka 2.4`), can avoid being affected using their own pluggable component with the old API. Note that none of these changes should affect the compatibility of CC with the version of managed Kafka cluster.

**Comment by akatona84** [comment]
@efeg  Could you check this PR again, please?

**Comment by viktorsomogyi** [comment]
Thanks @akatona84!

@efeg I checked the release notes of the 2.4 release and I don't think there is anything else that would break this kind of backward compatibility that you mentioned. Would you please check this PR again once you have some time for this?

**Comment by viktorsomogyi** [comment]
Thanks @efeg, I rebased it and also changed the Kafka version to 2.4.1 as the first bugfix release came out since. Checked its release notes, there are a couple of bug fixes but I think they don't break anything related to backward compatibility.

**Pull request merged**